vllm/spec_decode/spec_decode_worker.py

```diff
@@ -446,28 +475,43 @@ def _verify_tokens(
         # Get proposed tokens.
         proposal_token_ids = proposals.proposal_token_ids[spec_indices]
 
-        accepted_token_ids = self.rejection_sampler(
-            target_probs=proposal_verifier_probs,
-            bonus_token_ids=bonus_token_ids,
-            draft_probs=proposal_probs,
-            draft_token_ids=proposal_token_ids,
-        )
-
         # Append output tokens from non-speculative sequences to
         # the accepted token ids tensor.
         non_spec_token_ids = non_spec_token_ids.expand(-1, max_proposal_len +
                                                        1).clone()
         non_spec_token_ids[:, 1:] = -1
+        accepted_token_ids = self._get_accepted_token_ids(
+            proposal_verifier_probs=proposal_verifier_probs, bonus_token_ids=bonus_token_ids,
+            proposal_probs=proposal_probs, proposal_token_ids=proposal_token_ids)
         accepted_token_ids = torch.cat(
             [accepted_token_ids, non_spec_token_ids])
         logprobs = proposal_scores.logprobs
-
         # Rearrange so that results are in the order of the original seq group
         # metadata.
         accepted_token_ids[original_indices] = accepted_token_ids.clone()
 
         return accepted_token_ids, logprobs
 
+    def _get_accepted_token_ids(self, proposal_verifier_probs: torch.Tensor,
+                                bonus_token_ids: torch.Tensor,
+                                proposal_probs: torch.Tensor,
+                                proposal_token_ids: torch.Tensor):
+        if isinstance(self.spec_decode_sampler, RejectionSampler):
+            accepted_token_ids = self.spec_decode_sampler(
+                target_probs=proposal_verifier_probs,
+                bonus_token_ids=bonus_token_ids,
+                draft_probs=proposal_probs,
+                draft_token_ids=proposal_token_ids,
+            )
+        else:
+            assert isinstance(self.spec_decode_sampler, TypicalAcceptanceSampler)
+            accepted_token_ids = self.spec_decode_sampler(
+                target_probs=proposal_verifier_probs,
+                bonus_token_ids=bonus_token_ids,
+                draft_token_ids=proposal_token_ids,
+            )
+        return accepted_token_ids
```

Comment:
+1 to making them meet the same interface (both take draft_token_probs, typical acceptance ignores them)

Reply:
Added an abstract forward method to the base SpecDecodeBaseSampler and implementing the same in both the samplers.